libs/payment/api/src/lib/payment-provider/payrexx-subscription-payment-provider.ts

```diff
@@ -369,15 +370,32 @@ export class PayrexxSubscriptionPaymentProvider extends BasePaymentProvider {
     }
   }
 
-  async webhookForPaymentIntent(props: WebhookForPaymentIntentProps): Promise<IntentState[]> {
+  async webhookForPaymentIntent(props: WebhookForPaymentIntentProps): Promise<WebhookResponse> {
     const intentStates: IntentState[] = []
 
     // Protect endpoint
     const apiKey = props.req.query.apiKey as string
-    if (!timeConstantCompare(apiKey, this.webhookSecret)) throw new Error('Invalid api key!')
```

Comment:
this function is now reused in multiple files, can be extracted to utils

Reply:
Ah, let me get back to it if we rework it.